packages/manager/modules/pci/src/projects/project/storages/databases/databases.controller.js

```diff
@@ -38,4 +38,62 @@ export default class {
     this.trackDatabases('table::options_menu::delete_database');
     this.goToDeleteDatabase(database);
   }
+
+  renameDatabase(database) {
+    this.trackDatabases('table::options_menu::rename_database');
+    this.goToEditName(database.id);
+  }
+
+  upgradeVersion(database) {
+    this.trackDatabases('table::options_menu::change_version');
+    this.goToUpgradeVersion(database.id);
+  }
+
+  upgradePlan(database) {
+    this.trackDatabases('table::options_menu::upgrade_plan');
+    this.goToUpgradePlan(database.id);
+  }
+
+  upgradeNode(database) {
+    this.trackDatabases('table::options_menu::upgrade_node');
+    this.goToUpgradeNode(database.id);
+  }
+
+  getLatestVersion(database) {
+    this.currentEngine = this.engines.find(
+      (engine) => engine.name === database.engine,
+    );
+    return this.currentEngine.getLatestVersion().version;
+  }
+
+  getLatestPlan(database) {
+    this.currentEngine = this.engines.find(
+      (engine) => engine.name === database.engine,
+    );
+    return this.currentEngine.getLatestPlan(database.version, database.region)
+      .name;
+  }
+
+  getHighestFlavor(database) {
+    this.currentEngine = this.engines.find(
+      (engine) => engine.name === database.engine,
+    );
+    return this.currentEngine.getHighestFlavor(
+      database.version,
+      database.region,
+      database.plan,
+    ).name;
+  }
+
+  getCurrentFlavor(database) {
+    this.currentEngine = this.engines.find(
```

Comment:
Is it possible to extract currentEngine only once instead of doing it in all the getter methods? this will improve performance drastically.

Reply:
Done with https://github.com/ovh/manager/commit/fcd5037286f4bf8daffb8e00ebf54c5ee561d9b3